frontend/javascripts/admin/tasktype/task_type_create_view.js

```diff
@@ -226,131 +225,144 @@ class TaskTypeCreateView extends React.PureComponent<Props, State> {
+                      <Checkbox>Merger Mode</Checkbox>
+                    </FormItem>
+                  </React.Fragment>
+                ) : null
+              }
             </FormItem>
 
-            <div
-              style={{
-                marginLeft: 24,
-                display: this.props.form.getFieldValue(
-                  "settings.resolutionRestrictionsForm.shouldRestrict",
-                )
-                  ? "block"
-                  : "none",
-              }}
+            <FormItem
+              name={["settings", "resolutionRestrictionsForm", "shouldRestrict"]}
+              valuePropName="checked"
+              style={{ marginBottom: 6 }}
             >
-              <div>
-                <FormItem hasFeedback style={{ marginBottom: 6 }}>
-                  Minimum:{" "}
-                  {getFieldDecorator("settings.resolutionRestrictionsForm.min", {
-                    rules: [{ validator: isValidMagnification }],
-                  })(<InputNumber min={1} size="small" disabled={isEditingMode} />)}
-                </FormItem>
-              </div>
-              <div>
-                <FormItem hasFeedback>
-                  Maximum:{" "}
-                  {getFieldDecorator("settings.resolutionRestrictionsForm.max", {
-                    rules: [{ validator: isValidMagnification }],
-                  })(<InputNumber min={1} size="small" disabled={isEditingMode} />)}
-                </FormItem>
-              </div>
-            </div>
+              <Checkbox disabled={isEditingMode}>
+                Restrict Resolutions{" "}
+                <Tooltip
+                  title="The resolutions should be specified as power-of-two numbers. For example, if users should only be able to trace in the best and second best magnification, the minimum should be 1 and the maximum should be 2. The third and fourth resolutions can be addressed with 4 and 8."
+                  placement="right"
+                >
+                  <InfoCircleOutlined />
+                </Tooltip>
+              </Checkbox>
+            </FormItem>
+
+            <FormItem
+              noStyle
+              shouldUpdate={(prevValues, curValues) =>
+                !prevValues.settings ||
+                prevValues.settings.resolutionRestrictionsForm.shouldRestrict !==
+                  curValues.settings.resolutionRestrictionsForm.shouldRestrict
+              }
+            >
+              {({ getFieldValue }) =>
+                getFieldValue(["settings", "resolutionRestrictionsForm", "shouldRestrict"]) ? (
+                  <div style={{ marginLeft: 24 }}>
+                    <FormItem
+                      name={["settings", "resolutionRestrictionsForm", "min"]}
+                      hasFeedback
+                      label="Minimum"
+                      style={{ marginBottom: 6 }}
+                      rules={[{ validator: isValidMagnification }]}
```

Comment:
Maybe this is where the bug in this view is from / happening. For more, see my comment that lists the broken views.

Reply:
The bug was already resolved